org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/ResourceSetBasedAllContainersState.java

```diff
@@ -84,7 +84,7 @@ public String toString() {
 				result.append("(empty)");
 			else {
 				result.append("{\n    ");
-				result.append(Joiner.on("\n    ").join(uris));
+				result.append(uris.stream().map(URI::toString).collect(Collectors.joining("\n    ")));
```

Comment:
Quite a mouthful compared to Joiner.on. Both in terms of characters, cognitive load and object allocations. I'd keep this as a Joiner.

Reply:
> Quite a mouthful compared to Joiner.on. Both in terms of characters, cognitive load and object allocations.

Yes it has more characters, but personally I would say it is simple since you can read the pipeline from left to right, but I'm more used to Streams than Guava's Joiner so it is a matter of taste.
I haven't counted all object creations, but Joiner also creates objects and the JVM+GCs are usually good in handling short living objects respectively even avoiding their creation on the heap.
But the biggest impact is probably due to the creation of the String result of the join operation and if you are really concerned about object creation here, I would try to avoid that by using something like the following instead:
```
for (Iterator<String> containers = invisibleContainers.iterator(); containers.hasNext();) {
	result.append(containers.next());
	if(containers.hasNext()) {
		result.append("\n    ");
	}
}
```
That being said, I reverted that change for now.